src/pages/search/index.js

```diff
@@ -257,27 +257,21 @@ Search.getInitialProps = async (ctx) => {
       }
     }
   } = options
-
   const filterOrders = query.filter_order ? query.filter_order.split(',') : []
-  const queryString = {
+  const queryParams = {
     ...getSearchQueryForAPI(query),
-    limit: query.size || Number(limit),
-    offset: (query.p - 1 || Number(offset)) * (query.size || Number(limit)),
+    limit: query.limit || Number(limit),
+    offset: query.offset || Number(offset) * (query.limit || Number(limit)),
     ordering: query.sortby || ordering,
     ...(query.search ? { search: query.search } : {}),
     num_downloadable_samples__gt: !query.empty
       ? Number(numDownloadableSamples.hide)
       : Number(numDownloadableSamples.show)
   }
```

Comment:
I'll file an issue to support legacy URLs and clean this code block with a new helper method for a future sprint item 👍